scripts/test/run.py

```diff
@@ -14,7 +13,7 @@ def pyright_src():
 
 
 def pytest():
-    cmd = ["pytest"]
+    cmd = ["pytest", "--cache-clear"]
```

Comment:
double checking that not clearing the cache will always work.

Reply:
Locally, I do not need to clear the cache. Happy to remove it. The CI will anyways have a clean plate every time it runs.

Reply:
So I see.  Now your proposal is to *always* clear the cache.  I think that's a good idea.  Equivalent to what will happen with `mypy` where we *always* use `--no-incremental`